src/cell_group.hpp

```diff
@@ -127,22 +111,23 @@ class cell_group {
                           << gid_base_ << " at t " << cell_.time() << " ms\n";
             }
 
-            PE("events");
-            // check for new spikes
-            for (auto& s : spike_sources_) {
-                if (auto spike = s.source.test(cell_, cell_.time())) {
-                    spikes_.push_back({s.source_id, spike.get()});
-                }
-            }
-
             // apply events
+            PE("events");
             if (next) {
                 auto handle = get_target_handle(next->target);
                 cell_.deliver_event(handle, next->weight);
             }
             PL();
         }
 
+        // copy out voltage threshold crossings from the back end, then generate
```

Comment:
I had to read twice to understand, that you only copy the time for the cells that crossed. Rewording would make it easier.